datafusion/physical-expr/src/expressions/case.rs

```diff
@@ -998,6 +1080,53 @@ mod tests {
         Ok(())
     }
 
+    #[test]
```

Comment:
I think it would be good to make sure we had a `slt` level test to cover this as well,

Maybe in 
https://github.com/apache/datafusion/blob/382bf4f3c7a730828684b9e4ce01369b89717e19/datafusion/sqllogictest/test_files/expr.slt

Or we could start adding a file just for CASE if we are about to spend a bunch of time optimizing it 🤔

Reply:
I willl add slt tests

Reply:
I am planning on more CASE optimizations so will create a separate file

Reply:
I added the slt test. This is my first time using slt and it is very cool